tests/vmi_lifecycle_test.go

```diff
@@ -940,68 +940,136 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 		Context("[Serial]with node feature discovery", func() {
 
 			var node *k8sv1.Node
-			var originalLabels map[string]string
+			var cpuModelSupportedNodes = make([]string, 0)
+			var cpuFeatureSupportedNodes = make([]string, 0)
+			var supportedCPU string
+			var supportedFeature string
+
+			var supportedKVMInfoFeature []string
+
+			enableHyperVInVMI := func(label string) v1.FeatureHyperv {
+				features := v1.FeatureHyperv{}
+				trueV := true
+				switch label {
+				case "vpindex":
+					features.VPIndex = &v1.FeatureState{
+						Enabled: &trueV,
+					}
+				case "runtime":
+					features.Runtime = &v1.FeatureState{
+						Enabled: &trueV,
+					}
+				case "reset":
+					features.Reset = &v1.FeatureState{
+						Enabled: &trueV,
+					}
+				case "synic":
+					features.SyNIC = &v1.FeatureState{
+						Enabled: &trueV,
+					}
+				case "frequencies":
+					features.Frequencies = &v1.FeatureState{
+						Enabled: &trueV,
+					}
+				case "reenlightenment":
+					features.Reenlightenment = &v1.FeatureState{
+						Enabled: &trueV,
+					}
+				}
+
+				return features
+			}
 
 			BeforeEach(func() {
 				nodes := tests.GetAllSchedulableNodes(virtClient)
 				Expect(nodes.Items).ToNot(BeEmpty(), "There should be some compute node")
 
 				node = &nodes.Items[0]
-				originalLabels = node.GetObjectMeta().GetLabels()
 
-				tests.EnableFeatureGate(virtconfig.CPUNodeDiscoveryGate)
-			})
+				supportedCPU = tests.GetSupportedCPUModels(*node)[0]
+				supportedFeature = tests.GetSupportedCPUFeatures(*node)[0]
 
-			AfterEach(func() {
-				Expect(err).ToNot(HaveOccurred())
-				labelBytes, err := json.Marshal(originalLabels)
-				Expect(err).ToNot(HaveOccurred())
+				for key, _ := range node.Labels {
+					if strings.Contains(key, services.NFD_KVM_INFO_PREFIX) &&
+						!strings.Contains(key, "tlbflush") &&
+						!strings.Contains(key, "ipi") &&
+						!strings.Contains(key, "synictimer") {
+						supportedKVMInfoFeature = append(supportedKVMInfoFeature, strings.TrimPrefix(key, services.NFD_KVM_INFO_PREFIX))
+					}
 
-				node, err = virtClient.CoreV1().Nodes().Patch(context.Background(), node.Name, types.StrategicMergePatchType,
-					[]byte(fmt.Sprintf(`{"metadata": { "labels": %s}}`, labelBytes)), metav1.PatchOptions{})
-				Expect(err).ToNot(HaveOccurred(), "Should patch node successfully")
+				}
+
+				for _, n := range nodes.Items {
+					for key, _ := range n.Labels {
+						if strings.Contains(key, services.NFD_CPU_FEATURE_PREFIX+supportedFeature) {
```

Comment:
don't we know the exact name?

Reply:
We do, issue is, that each time, the test can run on different node and list of supported cpu models / features depends on the node HW. So before tests the script has to check which cpu models / features are supported. I can update it, so it will run only once for all tests.